packages/framework/src/Model/Product/Brand/BrandDataFactory.php

```diff
@@ -51,9 +65,13 @@ public function createFromBrand(Brand $brand)
             $brandData->descriptions[$translation->getLocale()] = $translation->getDescription();
         }
 
-        foreach ($brandDomains as $brandDomain) {
+        foreach ($brand->getDomains() as $brandDomain) {
```

Comment:
Is it necessary to make the getter public? It might be better to keep the concept of BrandDomains inside the Brand itself. You can iterate over $this->domain->getAll(). See that you access the $brandDomain here only in order to get its domain ID.

Reply:
I agree, now are all domain ids looped in this foreach